### PR TITLE
Baseline

### LINKED ISSUE #7: Add a simple baseline model
It is probably a good idea to have some kind of simple baseline that we can use to make sure the other components of the project are working, e.g. an LSTM or a 1D CNN.

### PR DESCRIPTION
**Pull request by disiji**
closes #7

### CODE REVIEW

**Comment by rloganiv** [comment]
closes #7

File: squawkbox/models/baseline.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            lstm_output, _ = self.lstm(packed, hidden)
+        # pad_packed_sequence
+        lstm_output, _ = pad_packed_sequence(lstm_output, batch_first=True)
+
+        # output = output.contiguous()
+        logits = self.h2o(lstm_output) * masks.unsqueeze(-1) # batch_size, seq_len, voc_len
+        output = {'logits': logits}
+
+        if tgt is not None:
+            ll = (F.log_softmax(logits, dim=-1).gather(2,tgt.unsqueeze(-1)) * masks.unsqueeze(-1)).mean(dim=1).mean(dim=0)
```
Comment (rloganiv) [inline squawkbox/models/baseline.py]: Separate this out into multiple lines so it is easier to read.

File: squawkbox/models/baseline.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        packed = pack_padded_sequence(embeddings, lengths=seq_lens, batch_first=True)
+        if hidden is None:
+            lstm_output, _ = self.lstm(packed)
+        else:
+            lstm_output, _ = self.lstm(packed, hidden)
+        # pad_packed_sequence
+        lstm_output, _ = pad_packed_sequence(lstm_output, batch_first=True)
+
+        # output = output.contiguous()
+        logits = self.h2o(lstm_output) * masks.unsqueeze(-1) # batch_size, seq_len, voc_len
```
Comment (rloganiv) [inline squawkbox/models/baseline.py:56]: After our discussion earlier, I realized this masking step is unneccessary. `pad_packed_sequences` will take care of masking for you.

File: squawkbox/models/baseline.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        self.lstm = nn.LSTM(
+            input_size=embedding_dim,
+            hidden_size=num_lstm_units,
+            num_layers=num_lstm_layers,
+            batch_first=True
+        )
+
+        self.h2o = nn.Linear(num_lstm_units, vocab_size)
+
+    def forward(self, src, tgt=None, hidden=None):
```
Comment (rloganiv) [inline squawkbox/models/baseline.py]: Please add `**kwargs` to the parameters. There are some additional inputs (related to keeping track of time) that will be fed into other models. By adding `**kwargs` you will allow `Baseline` to ignore these other inputs.

File: tests/test_baseline.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+
+def test_baseline():
+    baseline = Baseline(vocab_size=7, embedding_dim=2, num_lstm_units=2, num_lstm_layers=2)
+    src = torch.tensor([[1,2,3,0],[3,4,5,6]])
+    tgt = torch.tensor([[2,3,0,0],[4,5,6,1]])
+    out_dict = baseline(src=src, tgt=tgt)
+    out_dict['loss'].backward()
+    for p in baseline.parameters():
+        assert p.grad is not None
```
Comment (rloganiv) [inline tests/test_baseline.py:12]: Also add a test to make sure that `out_dict['logits']` is being properly masked for the first sequence.

**Review by rloganiv — CHANGES_REQUESTED** [review]

File: squawkbox/models/baseline.py (hunk 2 above)
Comment (rloganiv) [reply squawkbox/models/baseline.py:56]: Nvm

**Review by rloganiv — APPROVED** [review]
